app/models/conversion_host.rb

```diff
@@ -241,20 +241,27 @@ def resource_supports_conversion_host
   end
 
   # Find the credentials for the associated resource. By default it will
-  # look for a v2v auth type. If that is not found, it will look for the
-  # authentication associated with the resource using ssh_keypair or default,
-  # in that order, as the authtype.
+  # look for a v2v auth type if no argument is passed in. If that is not found,
+  # it will look for the first associated authentication. If one still isn't
+  # found, then it will look for the authentication associated with the resource
+  # using the 'ssh_keypair' auth type, and finally 'default'.
   #
-  def find_credentials(msg = nil)
-    authentication = authentication_type('v2v') ||
-      resource.authentication_type('ssh_keypair') ||
-      resource.authentication_type('default')
+  def find_credentials(auth_type = nil)
+    authentication = authentication_type(auth_type) || authentication_type('v2v') || authentications.first
+
+    if authentication.blank?
+      if resource.respond_to?(:authentication_type)
+        authentication = resource.authentication_type('ssh_keypair') || resource.authentication_type('default')
+      else
+        ems = resource.ext_management_system
+        authentication = ems.authentication_type('ssh_keypair') || ems.authentication_type('default')
```

Comment:
This could be simplified to `res = resource.respond_to?(:authentication_type) ? resource : resource.ext_management_system`

Reply:
Done!